packages/strapi-bookshelf/lib/index.js

```diff
@@ -345,9 +398,11 @@ module.exports = function(strapi) {
                       } else {
                         switch (attribute.type) {
                           case 'text':
-                          case 'json':
                             type = 'text';
                             break;
+                          case 'json':
+                            type = definition.client === 'pg' ? 'jsonb' : 'text';
```

Comment:
I recommend changing this to `longtext` for mysql per this doc by mariadb:

https://mariadb.com/kb/en/library/json-data-type/

In this case the `json` datatype is just an alias for longtext but the data storage is far greater.
`text` is limited to about ~ 16 Mb
`longtext` is limited to about ~ 4 Gb

Reply:
Here is the relevant doc for mysql as well, due to the differences in MariaDB `longtext` is likely the best option for use with both.

https://dev.mysql.com/doc/refman/5.7/en/json.html

Reply:
I will take a look today.

Reply:
No problem @lauriejim I was going to clone this locally and test the fix myself also. Along with my change.

Reply:
So far yours works fine, with the exception of the rendering in the adminUI (see example 1 below)
I believe this should be fixed with @soupette's new json editor. As it is now properly storing the data in MySQL (See example 2)

Example 1: 
![image](https://user-images.githubusercontent.com/8593673/41814459-b86614a6-7701-11e8-907a-bb0c1b5b2d95.png)

Example 2: 

![image](https://user-images.githubusercontent.com/8593673/41814461-c934679c-7701-11e8-832a-e55031fde80b.png)

After applying my change (changing the mysql type from `text` to `longtext`, it works as well with the exception of allowing for much larger blocks of json data.

@lauriejim if your testing also goes through can this be moved to Strapi 12.6 to go with the json input improvements?

Reply:
@lauriejim Awesome thanks, this helps move up some of my development time table :)